engine/core/Engine.cpp

```diff
@@ -230,3 +236,28 @@ bool Engine::isStarted() {
 double Engine::getDeltaTime() {
     return Engine::currentTime - Engine::lastTime;
 }
+
+std::string Engine::getConfigDir() {
+    // get the home directory and load the cache from there
+    // Here is my most convoluted ifdef set ever
+    std::string pathString;
+#ifndef _WIN32
+    const char *homePath = getenv("HOME");
+    if (!homePath) {
+        struct passwd* pwd = getpwuid(getuid());
+        if (pwd)
+            homePath = pwd->pw_dir;
+    }
+    pathString.append(homePath);
+#if defined(__APPLE__) && defined(__MACH__) // Assume macOS
+    pathString.append("/Library/Application Support/ChiraEngine/");
+#else // Assume Linux
+    pathString.append("/.local/ChiraEngine");
```

Comment:
What if someone runs two games made with the engine on their computer?

Reply:
this will be accounted for when the game info system is implemented. right now we have literally no method of detecting what game is being run properly outside of hardcoded things inside the game's code which is a horrible idea.

the plan is to add a json based source-esque gameinfo system and we'll determine the folder name from the internal game name/resource folder name stored in said file